### DISCUSSION TITLE
[2.9] Need to use Space or Enter to get to first submenu

### DISCUSSION THREAD

**Discussion by geekscrapy** [saulpw/visidata#1264] [General]
**Small description**
![Screen Shot 2022-01-10 at 20 29 50](https://user-images.githubusercontent.com/11225502/148801956-9b96aea6-7cba-4498-a186-dfc61198894c.png)

Using the keyboard, you cannot get to the sub items of "Plot > Graph". It just goes to the next main menu item, "System"

**Expected result**
To be able to get to "Graph"s sub menu items using the keyboard.

**Steps to reproduce with sample data and a .vd**
Open vd, activate the menu, scroll to Plot menu item, try and go right to select the Graph sub items

**Comment by saulpw** [comment]
Hi @geekscrapy, you can use `Space` or `Enter` to go into the first submenu.  I know it is a bit odd for the Right Arrow to act differently for the first submenu, but we had the behavior you're asking for originally, and it felt weirder to me.  Feel free to start a discussion (we can even convert this issue) if you feel strongly and want try to garner additional support for your position.

**Comment by geekscrapy** [comment]
Let's convert this to a discussion if you don't mind.

I don't think its intuitive for that at the moment. If the other submenu items required Space, or Enter, that's maybe a good middle ground, but at the moment they dont.

I think I get the reason for not auto "diving", but in this case I think it's counter intuitive, especially given behaviour of other GUI apps (Firefox's Toolbars menu has a submenu as its first option - it drops into the submenu right away, an extra right arrow and it moves to the next main menu item.

Another option might be to highlight (no drop-down) the main menu item when the ALT shortcut is pressed. The user could then cycle through the main menu items without them dropping into a submenu item. This does have the downside of having to then press down, Space or Enter to show the menu, but it might be faster?

**Comment by daviewales** [comment]
My 2 cents. Every other menu doesn't have this issue, _because the first item in every other menu doesn't point to a submenu_.

Why not just prevent the first item of the Plot menu from pointing to a submenu?

This could be easily achieved by moving 'graph current column' and 'graph all numeric columns' up a level directly into the top level Plot menu. e.g.:

```
Plot
│ current column           . │
│ all numeric columns     g. │
└────────────────────────────┘
```

**Reply by geekscrapy** [reply]
I did notice this subtlety also while looking for other examples! Have a look at Firefox though, it has the toolbars menu, and see what you think about that behaviour

**Reply by daviewales** [reply]
What OS are you on? My Firefox uses the MacOS global menubar, so I'm not sure I'll get the behaviour you're seeing.

However, this raises an interesting point.

On MacOS, when you select a menu, it only selects the menu title, _not the menu items_.
So, you can switch between menus using the left or right arrows, and the menu is shown, but no items are ever selected unless you use the up or down arrows.

<img width="212" alt="Screen Shot 2022-01-18 at 5 29 20 pm" src="https://user-images.githubusercontent.com/2499066/149882825-8020018e-f5b4-4e36-b2ea-651e6014e277.png">

I think this is probably the best option!

**Reply by geekscrapy** [reply]
Yea, I would agree
